src/Umbraco.Infrastructure/Migrations/Upgrade/V_16_3_0/MigrateMediaTypeLabelProperties.cs

```diff
@@ -0,0 +1,199 @@
+            return;
+        }
+
+        var nodeDto = new NodeDto
+        {
+            NodeId = Constants.DataTypes.LabelPixels,
+            Trashed = false,
+            ParentId = -1,
+            UserId = -1,
+            Level = 1,
+            Path = "-1," + Constants.DataTypes.LabelPixels,
+            SortOrder = 41,
+            UniqueId = _labelPixelsDataTypeKey,
+            Text = "Label (pixels)",
+            NodeObjectType = Constants.ObjectTypes.DataType,
+            CreateDate = DateTime.Now,
+        };
+
+        _ = Database.Insert(Constants.DatabaseSchema.Tables.Node, "id", false, nodeDto);
+
+        var dataTypeDto = new DataTypeDto
+        {
+            NodeId = Constants.DataTypes.LabelPixels,
+            EditorAlias = Constants.PropertyEditors.Aliases.Label,
+            EditorUiAlias = "Umb.PropertyEditorUi.Label",
+            DbType = nameof(ValueStorageType.Integer),
+            Configuration = "{\"umbracoDataValueType\":\"INT\", \"labelTemplate\":\"{=value}px\"}",
+        };
+
+        _ = Database.Insert(Constants.DatabaseSchema.Tables.DataType, "pk", false, dataTypeDto);
+    }
+
+    private async Task MigrateMediaTypeLabels()
+    {
+        // update the media types with the new data-type references
+        foreach (string mediaTypeKey in _mediaTypeKeys)
+        {
+            if (_mediaTypeSettings?.InstallData == InstallDefaultDataOption.Values && !_mediaTypeSettings.Values.InvariantContains(mediaTypeKey))
+            {
+                continue;
+            }
+
+            if (_mediaTypeSettings?.InstallData == InstallDefaultDataOption.ExceptValues && _mediaTypeSettings.Values.InvariantContains(mediaTypeKey))
+            {
+                continue;
+            }
+
+            IMediaType? mediaType = await _mediaTypeService.GetAsync(new Guid(mediaTypeKey));
+
+            if (mediaType is null)
+            {
+                continue;
+            }
+
+            bool updated = false;
+
+            foreach (IPropertyType propertyType in mediaType.PropertyTypes)
+            {
+                switch (propertyType.Alias)
+                {
+                    case Constants.Conventions.Media.Bytes:
+                        if (propertyType.DataTypeId == Constants.DataTypes.LabelBigint)
+                        {
+                            propertyType.DataTypeId = Constants.DataTypes.LabelBytes;
+                            propertyType.DataTypeKey = _labelBytesDataTypeKey;
+                            updated = true;
+                        }
+
+                        break;
+
+                    case Constants.Conventions.Media.Height:
+                    case Constants.Conventions.Media.Width:
+                        if (propertyType.DataTypeId == Constants.DataTypes.LabelInt)
+                        {
+                            propertyType.DataTypeId = Constants.DataTypes.LabelPixels;
+                            propertyType.DataTypeKey = _labelPixelsDataTypeKey;
```

Comment:
Should we align the description here?

```suggestion
                            propertyType.DataTypeKey = _labelPixelsDataTypeKey;

                            if (propertyType.Description == "in pixels")
                            {
                                propertyType.Description = null;
                            }
```